ryhope/src/storage/mod.rs

```diff
@@ -135,6 +148,124 @@ impl<K: Debug + Hash + Eq + Clone + Sync + Send, V: Clone> WideLineage<K, V> {
     }
 }
 
+// An `EpochMapper` allows to map `UserEpoch` to `IncrementalEpoch` of
+// a `TreeStorage`, and vice versa
+pub trait EpochMapper: Sized + Send + Sync + Clone + Debug {
+    fn try_to_incremental_epoch(
+        &self,
+        epoch: UserEpoch,
+    ) -> impl Future<Output = Option<IncrementalEpoch>> + Send;
+
+    fn to_incremental_epoch(
+        &self,
+        epoch: UserEpoch,
+    ) -> impl Future<Output = IncrementalEpoch> + Send {
+        async move {
+            self.try_to_incremental_epoch(epoch)
+                .await
+                .unwrap_or_else(|| panic!("IncrementalEpoch corresponding to {epoch} not found"))
+        }
+    }
+
+    fn try_to_user_epoch(
+        &self,
+        epoch: IncrementalEpoch,
+    ) -> impl Future<Output = Option<UserEpoch>> + Send;
+
+    fn to_user_epoch(&self, epoch: IncrementalEpoch) -> impl Future<Output = UserEpoch> + Send {
+        async move {
+            self.try_to_user_epoch(epoch)
+                .await
+                .unwrap_or_else(|| panic!("UserEpoch corresponding to {epoch} not found"))
+        }
+    }
+
+    fn add_epoch_map(
+        &mut self,
+        user_epoch: UserEpoch,
+        incremental_epoch: IncrementalEpoch,
+    ) -> impl Future<Output = Result<()>> + Send;
+}
+#[derive(Clone, Debug)]
+pub struct SharedEpochMapper<T: EpochMapper, const READ_ONLY: bool>(Arc<RwLock<T>>);
```

Comment:
I do not understand the purpose of this flag, that is interpreted as `EXTERNAL_EPOCH_MAPPER` later.

Reply:
Added comments to explain better the purpose of this flag in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/425/commits/23affb5f3b0c62fc1ab825043196a2f89e20039c